### DISCUSSION TITLE
In a Shiny app, is there a way to capture the insight::format_alert text?

### DISCUSSION THREAD

**Discussion by ProfessorPeregrine** [easystats/bayestestR#657] [Q&A] [answered]
I have a large Shiny app and am adding some credible interval output via ci(). In some situations the script will use insight::format_alert() to send a message to the console, for example:

` 'ci' is too large or x does not contain enough data points, returning NAs.`

What I would like to do is capture any alert text and add it to the output in the Shiny app. As it is, it goes to the console, but the user might not see that and might not understand why there are no CIs.

I tried to use tryCatch() but it isn't working since I don't think the message counts as a warning or error, though maybe I am off-base about that.

Is this possible?

Thank you very much!

**Comment by bwiernik** [comment, marked as answer]
These are "messages" in R's exception lingo. Here is some code that you could use to capture messages, errors, and warnings 

https://www.r-bloggers.com/2020/10/capture-message-warnings-and-errors-from-a-r-function/

**Reply by ProfessorPeregrine** [reply]
Thank you, that was very helpful and I learned a couple of new things today!
